lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1694,17 +1694,26 @@ mkApiTransaction ti txid ins outs ws (meta, timestamp) setTimeReference = do
     tx :: ApiTransaction n
     tx = ApiTransaction
         { id = ApiT txid
-        , amount = meta ^. #amount
+        , amount = amountWithWithdrawals
         , insertedAt = Nothing
         , pendingSince = Nothing
         , depth = Nothing
         , direction = ApiT (meta ^. #direction)
         , inputs = [ApiTxInput (fmap toAddressAmount o) (ApiT i) | (i, o) <- ins]
         , outputs = toAddressAmount <$> outs
-        , withdrawals = mkApiWithdrawal @n <$> Map.toList ws
+        , withdrawals = toApiWithdrawal
         , status = ApiT (meta ^. #status)
         }
 
+    toApiWithdrawal :: [ApiWithdrawal n]
+    toApiWithdrawal = mkApiWithdrawal @n <$> Map.toList ws
+
+    amountWithWithdrawals :: Quantity "lovelace" Natural
+    amountWithWithdrawals =
+        let amtFromMeta = meta ^. #amount
+            amtFromWithdrawal = map (\(ApiWithdrawal _ amt) -> amt) toApiWithdrawal
+        in Quantity $ sum $ getQuantity <$>  (amtFromMeta : amtFromWithdrawal)
```

Comment:
On a second thought, what about doing this calculation earlier when the `meta ^. #amount` is actually first created? Otherwise, we get to carry this inconsistency everywhere instead of being solved in a more holistic manner.